lowranceusr.cc

```diff
@@ -423,28 +975,399 @@ lowranceusr_parse_waypt(Waypoint* wpt_tmp)
+    if (global_opts.debug_level == 99) {
+      printf(MYNAME " parse_waypoints: ");
+      if (reading_version > 4) {
+        printf("%08x %08x %08x %08x ",
+               fsdata->UUID1, fsdata->UUID2, fsdata->UUID3, fsdata->UUID4);
+      }
+      printf(" %10u %8d %8d %8d %6d %16s",
+             fsdata->uid_unit, fsdata->uid_seq_low, fsdata->uid_seq_high,
+             waypoint_version, name_len, name_buff);
+      if (reading_version > 4) {
+        printf("  %10u ", fsdata->uid_unit2);
+      }
+      printf(" %+15.10f %+15.10f", wpt_tmp->longitude, wpt_tmp->latitude);
+      printf(" %08x %4d %4d %7s", fsdata->flags, fsdata->icon_num, fsdata->color, qPrintable(fsdata->color_desc));
+      printf(" %6d %16s", desc_len, desc_buff);
+      QDateTime dt;
+      QTime t;
+      dt = wpt_tmp->GetCreationTime();
+      t  = wpt_tmp->GetCreationTime().time();
+      printf(" %s %s", qPrintable(dt.toString("yyyy/MM/dd")), qPrintable(t.toString("hh:mm")));
+      printf(" %08x %f %08x %08x %08x\n",
+             unused_byte, wpt_tmp->altitude, loran_GRI, loran_Tda, loran_Tdb);
+    } else {
+      printf(MYNAME " parse_waypoints: version = %d, name = %s, uid_unit = %u, "
+             "uid_seq_low = %d, uid_seq_high = %d, lat = %+.10f, lon = %+.10f, depth = %f\n",
+             waypoint_version, qPrintable(wpt_tmp->shortname), fsdata->uid_unit,
+             fsdata->uid_seq_low, fsdata->uid_seq_high,
+             wpt_tmp->longitude, wpt_tmp->latitude, wpt_tmp->altitude);
+    }
+  }
+}
 
 
 static void
-lowranceusr_parse_routes()
+lowranceusr_parse_waypts()
+{
+  int NumWaypoints;
+
+  if (reading_version < 4) {
+    /* USR versions 2 & 3 have a 16 bit count */
+    NumWaypoints = gbfgetint16(file_in);
+  } else {
+    /* Starting with USR version 4 have 32 bit count */
+    NumWaypoints = gbfgetint32(file_in);
+  }
+
+  if (global_opts.debug_level >= 1) {
+    printf(MYNAME " parse_waypts: Num Waypoints = %d\n", NumWaypoints);
+  }
+
+  if (global_opts.debug_level == 99) {
+    if (reading_version > 3) {
+      printf(MYNAME " parse_waypts: ");
+      if (reading_version > 4) {
+        printf("Universal ID                        ");
+      }
+      printf("              Sequence Number  Stream  Waypoint\n");
+
+      printf(MYNAME " parse_waypoints: ");
+      if (reading_version > 4) {
+        printf("    ID1      ID2      ID3      ID4  ");
+      }
+      printf("Unit Number     Low      High  Version Length Name            ");
+      if (reading_version > 4) {
+        printf(" Unit Number2");
+      }
+      printf(" Latitude        Longitude       Flags    ICON Color        Length Description     ");
+      printf(" Date       Time  Unused   Depth    LoranGRI LoranTda LoranTdb\n");
+
+      printf(MYNAME " parse_waypoints: ");
+      if (reading_version > 4) {
+        printf("-------- -------- -------- -------- ");
+      }
+      printf("----------- -------- -------- -------- ------ ----------------");
```

Comment:
This will admittedly be difficult/fragile with qDebug(), but it's probably fragile here, too. Maybe it all becomes a qtextstream and http://doc.qt.io/qt-5/qtextstream.html#setFieldWidth and let that handle the columnization.